### ISSUE TITLE
Differences between prediction and label types.

### ISSUE THREAD

**Issue by Jarartur** [QiLi111/tus-rec-challenge_baseline#4]
Hi. Thank you for this framework to work with during the challenge.

I am not sure though what each type is supposed to be exactly. I went through the code and have a high-level understanding, but not the details. 

Based on the documentation and code, I figured that:
- `transformation` is an affine matrix (size=Nx4x4), 
- `parameters` are (rx, ry, rz, tx, ty, tz) (size=Nx6)
- `points` are landmarks (x, y, z) (size=Nx3)

But now I fall short of understanding the default trainloop. If during the training we load ground truth labels of type `transform` *(because `tforms` are of shape 4x4 and are transformation matrices)*, why is it specified that the labels are of type `points` in the default `TrainOptions`? It impacts the `criterion` calculations as well as `metrics` later on.

**Comment by QiLi111** [comment]
Hi,

Although the initial ground truth is defined in transformation matrix, we could convert this into other types, for example, [parameter](https://github.com/QiLi111/tus-rec-challenge_baseline/blob/a4dbbabf057573ad9982c31cc2cbf2eabeb943e2/utils/transform.py#L106), or [points](https://github.com/QiLi111/tus-rec-challenge_baseline/blob/a4dbbabf057573ad9982c31cc2cbf2eabeb943e2/utils/transform.py#L89). The prediction could be in any type as well (either parameter, transformation, or point). When calculating loss or metrics, the types of the two must be the same, for example, in this baseline code, we convert the type of prediction into the type of label (see and example [here](https://github.com/QiLi111/tus-rec-challenge_baseline/blob/a4dbbabf057573ad9982c31cc2cbf2eabeb943e2/utils/transform.py#L164)). 

We provide various combinations of prediction and label types, listing all possible scenarios. For example, if both `PRED_TYPE` and `LABEL_TYPE` are set as "transformation", then, the network will predict transformation matrix, and label is the initial ground truth (transformation matrix). No transformation operation is needed in this case. If `PRED_TYPE` is "transformation", and `LABEL_TYPE` is "point", then the network will predict transformation matrix, and then converted into [points](https://github.com/QiLi111/tus-rec-challenge_baseline/blob/a4dbbabf057573ad9982c31cc2cbf2eabeb943e2/utils/transform.py#L251), and the labels will be [points](https://github.com/QiLi111/tus-rec-challenge_baseline/blob/a4dbbabf057573ad9982c31cc2cbf2eabeb943e2/utils/transform.py#L89) as well, converted from initial transformation matrix.

Note that the purpose of training loop is to train a model, and during inference/testing, the algorithm is expected to take the entire scan as input and output four different sets of transformation-representing displacement vectors as results, a set of displacement vectors on individual pixels and a set of displacement vectors on provided landmarks, on global level and local level, respectively. (see [this](
https://github.com/QiLi111/tus-rec-challenge_baseline/blob/a4dbbabf057573ad9982c31cc2cbf2eabeb943e2/utils/evaluation.py#L182) for an example). We will also provide detailed submission instructions regarding the shapes of the prediction etc.

Hope this makes sense.

Best regards,
Qi

**Comment by QiLi111** [comment]
`reference_image_points` denotes the coordinates of points in the image coordinate system, which is consist with the coordinates system during calibration process.
